modules/apps/foundation/user-associated-data/user-associated-data-api/src/main/java/com/liferay/user/associated/data/model/impl/UADEntityExporterImpl.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.user.associated.data.model.impl;
+
+import com.liferay.user.associated.data.model.UADEntity;
+import com.liferay.user.associated.data.model.UADEntityExporter;
+
+import java.util.List;
+
+/**
+ * @author William Newbury
+ */
+public abstract class UADEntityExporterImpl implements UADEntityExporter {
+
+	@Override
+	public abstract void export(UADEntity uadEntity);
+
+	@Override
+	public void exportAll(long userId) {
+		for (UADEntity uadEntity : getUADEntities(userId)) {
```

Comment:
This doesn't scale, because there could be thousands or even millions of entities.